packages/appkit/src/client.ts

```diff
@@ -1282,10 +1283,12 @@ export class AppKit {
               chainId: ChainController.state.activeCaipNetwork?.id as string | number
             })
           } catch (error) {
-            adapter?.switchNetwork({
-              provider: this.universalProvider,
-              caipNetwork: ChainController.state.activeCaipNetwork as CaipNetwork
-            })
+            if (adapter?.reconnect) {
+              adapter?.reconnect({
+                id: 'walletConnect',
+                type: 'WALLET_CONNECT'
+              })
```

Comment:
can we add a comment on why this is required? not sure why this or a switch network would be required if a connection fails? 🤔

Reply:
Added a comment:

`Handle edge case where wagmi detects existing connection but lacks to complete UniversalProvider instance. Connection attempt fails due to already connected state - reconnect to restore provider state.`